gobblin-core/src/main/java/gobblin/source/extractor/watermark/TimestampWatermark.java

```diff
@@ -46,6 +47,7 @@
   public TimestampWatermark(String watermarkColumn, String watermarkFormat) {
     this.watermarkColumn = watermarkColumn;
     this.watermarkFormat = watermarkFormat;
+    _inputFormatParser = new SimpleDateFormat(INPUTFORMAT);
```

Comment:
The line affected in `Partitioner` is:
`private static final SimpleDateFormat WATERMARKTIMEPARSER = new SimpleDateFormat(WATERMARKTIMEFORMAT);`